pkg/controller/node/nodecontroller.go

```diff
@@ -212,7 +213,11 @@ func (nc *NodeController) generateAvailableCIDRs() {
 	// Generate all available CIDRs here, since there will not be manay
 	// available CIDRs. Set will be small, it will use less than 1MB memory
 	cidrIP := nc.clusterCIDR.IP.To4()
-	nc.maxCIDRs = (256-int(cidrIP[1]))*256 - int(cidrIP[2])
+
+	cidrSize, _ := nc.clusterCIDR.Mask.Size()
+
+	// nc.maxCIDRs = int(math.Mod((256-cidr[0])*2^24 + (256-cidr[1])*2^16 + (256-cidr[2])*2^8, 2^(32-cidrSize)) / (2^8))
```

Comment:
I think the old code let you allocate subnets from some point in the middle of the CIDR and only go forward, which might be neat for some, but does not really meet the established definition of which range CIDR notation represents exactly. :-) I'm assuming this was an attempt to replicate that non-standard behaviour. #19242 is the way to go, as it does things the proper way and supports non-/24 pod allocations.